homeassistant/components/scene.py

```diff
@@ -178,14 +179,38 @@ def _state_as_requested(self, cur_state):
         """ Returns if given state is as requested. """
         state = self.scene_config.states.get(cur_state and cur_state.entity_id)
 
-        return (cur_state is not None and state.state == cur_state.state and
+        ret = (cur_state is not None and state.state == cur_state.state and
                 all(value == cur_state.attributes.get(key)
                     for key, value in state.attributes.items()))
 
+        print('AS REQUESTED')
+        print(ret)
+        print(self.scene_config.states)
+        return ret
+
+    def _state_read_after_set(self, states):
```

Comment:
Now it seems that you are changing the actual config passed in, that seems very odd to me. Wouldn't it be easier to just record the states again so it stays on?
```python
        self.prev_states = tuple(self.hass.states.get(entity_id)
                                 for entity_id in self.entity_ids)
```